pyro/infer/map.py

```diff
@@ -0,0 +1,22 @@
+from pyro.infer.svi import SVI
+
+
+def MAP(model, optim, *args, **kwargs):
+    """
+    :param model: the model (callable containing Pyro primitives)
+    :param optim: a wrapper a for a PyTorch optimizer
+    :type optim: pyro.optim.PyroOptim
+
+    Maximum A Posteriori inference.
+
+    MAP inference is useful for estimating parameters of models that have no
+    latent variables, or where all latent variables have been replaced by
+    parameters. In MAP models, the likelihood is specified by ``pyro.observe``
+    statements that observe evidence, and the prior is specified by
+    ``pyro.observe`` statements that observe parameter values.
+    """
+
+    def guide(*args, **kwargs):
+        pass
+
+    return SVI(model, guide, optim, loss="ELBO", *args, **kwargs)
```

Comment:
this could be a one liner right:
```
return SVI(model, lambda *args, **kwargs: None, optim, loss="ELBO", *args, **kwargs)
```
though im sure theres some SO post that says its more pythonic to do it your way

Reply:
Right, but I think it's easier to understand when we write out the trivial guide.

Reply:
Is it better to define MAP as a class (as a sub-class of SVI)? I know the returned value would be the same. But with class, we can inherit to define other inference such as Quadratic approximation (i.e. Laplace approximation), which requires computing hessian after getting the MAP.